### ISSUE TITLE
Migration script generates wrong sentry_rule.data values

### ISSUE THREAD

**Issue by eagletmt** [getsentry/sentry#1365]
I tried to upgrade from Sentry 6.4.4 to 7.0.1 and found that `sentry_rule.data` column is corrupted during the migration in `sentry --config=/etc/sentry.conf.py upgrade`.

The wrong values are only generated during the migration.
When I create a new project, the auto-generated `sentry_rule.data` values look good.

The following data is my `sentry_rule` table.
project_id = 1, 2, 3 are migrated projects and project_id = 4 is newly created project after upgrading.

```
sentry=> SELECT * FROM sentry_rule;
 id | project_id |                label                |                                                                                             data                                                                                             |          date_added
----+------------+-------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------
  1 |          2 | Send a notification for new events  | {'conditions': [{'id': 'sentry.rules.conditions.first_seen_event.FirstSeenEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}                     | 2015-01-03 10:32:07.179107+09
  2 |          2 | Send a notification for regressions | {'conditions': [{'id': 'sentry.rules.conditions.regression_event.RegressionEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}                    | 2015-01-03 10:32:07.180056+09
  3 |          3 | Send a notification for new events  | {'conditions': [{'id': 'sentry.rules.conditions.first_seen_event.FirstSeenEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}                     | 2015-01-03 10:32:07.180519+09
  4 |          3 | Send a notification for regressions | {'conditions': [{'id': 'sentry.rules.conditions.regression_event.RegressionEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}                    | 2015-01-03 10:32:07.180952+09
  5 |          1 | Send a notification for new events  | {'conditions': [{'id': 'sentry.rules.conditions.first_seen_event.FirstSeenEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}                     | 2015-01-03 10:32:07.181377+09
  6 |          1 | Send a notification for regressions | {'conditions': [{'id': 'sentry.rules.conditions.regression_event.RegressionEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}                    | 2015-01-03 10:32:07.181797+09
  7 |          4 | Send a notification for new events  | eJxljbsOwjAMRXf/SDJZlDcjQjCy5AOqKklRpMqJ4oDUv8epUjGw+drX52iXOjDKRnKhhEisIG1BT2kH2qW9nIKT1UEG9lTyjPk9ecbfA44hc+nZe+r9Rzr4qAsj+V7jbW0K5gg8sFGDXVWnqjpX1QVef5LWQ4oljHOjP5ewoK+2cbtNBeMX2PhGnQ== | 2015-01-03 10:41:17.829926+09
  8 |          4 | Send a notification for regressions | eJzTSCkw5ApWT87PS8ksyczPK1bnKjDi0sgpMObSSCkwAUplpgCFTIGM4tS8kqJKvaLSnNRiPYQGvaLU9KLU4mIgOz61DKhGLwgu4AriO8OUAs0x4ypOLA5WT0yG2WUOsssCZJclVzqGLVB1enn5JZlplVDj/cAcsNGOyVBzDQ1ABusBAD+xRzo=     | 2015-01-03 10:41:17.831573+09
(8 rows)
```

**Comment by eagletmt** [comment]
My quick fix is to to update values after upgrading :innocent: 
https://gist.github.com/eagletmt/0611d525df253362d4ac

**Comment by dcramer** [comment]
I think I have a quick fix for this, though it wont fix existing bad values. I have a feeling we did this migration manually which is why we didn't notice this with getsentry.com

**Comment by dcramer** [comment]
I'll plan to push out a bugfix release for this tomorrow when I have more time to QA it

**Comment by eagletmt** [comment]
I rolled back my database and retried `sentry upgrade` with the latest version 8dce0ab9c9ae94619d58f3ba89c1f88976c7a372 , but `sentry_rule.data` still becomes corrupted.

```
% psql sentry
psql (9.3.5)
Type "help" for help.

sentry=> SELECT * FROM sentry_rule;
ERROR:  relation "sentry_rule" does not exist
LINE 1: SELECT * FROM sentry_rule;
                      ^
sentry=> \q

% sentry --config=/etc/sentry.conf.py upgrade
Syncing...
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)
Migrating...
Running migrations for djcelery:
- Nothing to migrate.
 - Loading initial data for djcelery.
Installed 0 object(s) from 0 fixture(s)
Running migrations for django:
- Nothing to migrate.
 - Loading initial data for django.
Installed 0 object(s) from 0 fixture(s)
Running migrations for sentry:
 - Migrating forwards to 0149_auto__chg_field_groupseen_project__chg_field_groupseen_user__chg_field.
 > sentry:0109_index_filtervalue_times_seen
 > sentry:0110_index_filtervalue_last_seen
 > sentry:0111_index_filtervalue_first_seen
 > sentry:0112_auto__chg_field_option_value__chg_field_useroption_value__chg_field_pr
 > sentry:0113_auto__add_field_team_status
 > sentry:0114_auto__add_field_projectkey_roles
 > sentry:0115_auto__del_projectcountbyminute__del_unique_projectcountbyminute_projec
 > sentry:0116_auto__del_field_event_server_name__del_field_event_culprit__del_field_
 > sentry:0117_auto__add_rule
 > sentry:0118_create_default_rules
 > sentry:0119_auto__add_field_projectkey_label
 > sentry:0120_auto__add_grouprulestatus
 > sentry:0121_auto__add_unique_grouprulestatus_rule_group
 > sentry:0122_add_event_group_id_datetime_index
 > sentry:0123_auto__add_groupassignee__add_index_event_group_datetime
 > sentry:0124_auto__add_grouphash__add_unique_grouphash_project_hash
 > sentry:0125_auto__add_field_user_is_managed
 > sentry:0126_auto__add_field_option_last_updated
[WARNING] /home/sentry/.local/lib/python2.7/site-packages/django/db/models/fields/__init__.py:903: RuntimeWarning: DateTimeField FakeModel.last_updated received a naive datetime (2015-01-03 04:07:18.638354) while time zone support is active.
  RuntimeWarning)

 > sentry:0127_auto__add_release__add_unique_release_project_version
 > sentry:0128_auto__add_broadcast
 > sentry:0129_auto__chg_field_release_id__chg_field_pendingteammember_id__chg_field_
 > sentry:0130_auto__del_field_project_owner
 > sentry:0131_auto__add_organizationmember__add_unique_organizationmember_organizati
 > sentry:0132_add_default_orgs
Teams: 100% |###########################################################################################################################################################################| Time: 00:00:00
 > sentry:0133_add_org_members
Teams: 100% |###########################################################################################################################################################################| Time: 00:00:00
 > sentry:0134_auto__chg_field_team_organization
 > sentry:0135_auto__chg_field_project_team
 > sentry:0136_auto__add_field_organizationmember_email__chg_field_organizationmember
 > sentry:0137_auto__add_field_organizationmember_has_global_access
 > sentry:0138_migrate_team_members
Organizations: 100% |###################################################################################################################################################################| Time: 00:00:00
 > sentry:0139_auto__add_auditlogentry
 > sentry:0140_auto__add_field_organization_slug
 > sentry:0141_fill_org_slugs
Organizations: 100% |###################################################################################################################################################################| Time: 00:00:00
 > sentry:0142_auto__add_field_project_organization__add_unique_project_organization_
 > sentry:0143_fill_project_orgs
Projects: 100% |########################################################################################################################################################################| Time: 00:00:00
 > sentry:0144_auto__chg_field_project_organization
 > sentry:0145_auto__chg_field_organization_slug
 > sentry:0146_auto__add_field_auditlogentry_ip_address
 > sentry:0147_auto__del_unique_team_slug__add_unique_team_organization_slug
 > sentry:0148_auto__add_helppage
 > sentry:0149_auto__chg_field_groupseen_project__chg_field_groupseen_user__chg_field
The following content types are stale and need to be deleted:

    sentry | projectcountbyminute
    sentry | groupcountbyminute

Any objects related to these content types by a foreign key will also
be deleted. Are you sure you want to delete these content types?
If you're unsure, answer 'no'.

    Type 'yes' to continue, or 'no' to cancel: yes
 - Loading initial data for sentry.
Installed 0 object(s) from 0 fixture(s)
Running migrations for nodestore:
- Nothing to migrate.
 - Loading initial data for nodestore.
Installed 0 object(s) from 0 fixture(s)
Running migrations for search:
- Nothing to migrate.
 - Loading initial data for search.
Installed 0 object(s) from 0 fixture(s)
Running migrations for social_auth:
- Nothing to migrate.
 - Loading initial data for social_auth.
Installed 0 object(s) from 0 fixture(s)

Synced:
 > django.contrib.admin
 > django.contrib.auth
 > django.contrib.contenttypes
 > django.contrib.messages
 > django.contrib.sessions
 > django.contrib.sites
 > django.contrib.staticfiles
 > captcha
 > crispy_forms
 > raven.contrib.django.raven_compat
 > rest_framework
 > sentry.plugins.sentry_interface_types
 > sentry.plugins.sentry_mail
 > sentry.plugins.sentry_urls
 > sentry.plugins.sentry_useragents
 > south
 > sudo

Migrated:
 - djcelery
 - kombu.transport.django
 - sentry
 - sentry.nodestore
 - sentry.search
 - social_auth
Loading help page tagging.md
Loading help page teams_and_projects.md
Loading help page organizations.md
Loading help page sampling.md
Loading help page quotas.md

% psql sentry
psql (9.3.5)
Type "help" for help.

sentry=> SELECT * FROM sentry_rule;
 id | project_id |                label                |                                                                                   data                                                                                    |          date_added
----+------------+-------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------
  1 |          2 | Send a notification for new events  | {'conditions': [{'id': 'sentry.rules.conditions.first_seen_event.FirstSeenEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}  | 2015-01-03 13:07:18.067402+09
  2 |          2 | Send a notification for regressions | {'conditions': [{'id': 'sentry.rules.conditions.regression_event.RegressionEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]} | 2015-01-03 13:07:18.068403+09
  3 |          3 | Send a notification for new events  | {'conditions': [{'id': 'sentry.rules.conditions.first_seen_event.FirstSeenEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}  | 2015-01-03 13:07:18.068874+09
  4 |          3 | Send a notification for regressions | {'conditions': [{'id': 'sentry.rules.conditions.regression_event.RegressionEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]} | 2015-01-03 13:07:18.069331+09
  5 |          1 | Send a notification for new events  | {'conditions': [{'id': 'sentry.rules.conditions.first_seen_event.FirstSeenEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]}  | 2015-01-03 13:07:18.069763+09
  6 |          1 | Send a notification for regressions | {'conditions': [{'id': 'sentry.rules.conditions.regression_event.RegressionEventCondition'}], 'actions': [{'id': 'sentry.rules.actions.notify_event.NotifyEventAction'}]} | 2015-01-03 13:07:18.070197+09
(6 rows)
sentry=> \q
%
```

**Comment by eagletmt** [comment]
The error is reported by sentry worker.

```
[ERROR] Incorrect padding
Traceback (most recent call last):
File "/path/to/sentry/src/sentry/db/models/fields/gzippeddict.py", line 35, in to_python
value = pickle.loads(decompress(value))
File "/path/to/sentry/src/sentry/utils/strings.py", line 37, in decompress
return zlib.decompress(base64.b64decode(value))
File "/usr/lib/python2.7/base64.py", line 76, in b64decode
raise TypeError(msg)
TypeError: Incorrect padding

```

**Comment by dcramer** [comment]
Looks like we missed a significant amount of these.

**Comment by dcramer** [comment]
This is fixed in 7.0.2 (and should correct existing corrupted values)

**Comment by eagletmt** [comment]
Upgrading to 7.0.2 fixes this issue :ok_woman:
